### DISCUSSION TITLE
low free memory

### DISCUSSION THREAD

**Discussion by ManfredBartz** [arendst/Tasmota#16220] [Q&A] [answered]
While testing a new HA automation, I managed to brick a group of 6 downlights.  I can recover them by the 6 power-cycle method but I don't want to keep doing that too often...

Checking the info, it appears free memory is very low, 25.7kB.
Are there any options to free up some memory?

Program Version  11.0.0(tasmota)
ESP Chip Id   4958066 (ESP8266EX)
Flash Chip Id   0x1540A1
Flash Size	2048 kB
Program Flash Size	1024 kB
Program Size	620 kB
Free Program Space	380 kB
Free Memory	25.7 kB

**Comment by barbudor** [comment, marked as answer]
How do you qualify "low memory"?
25kB is 31% of total RAM
I have a few Esp8266 devices running stable with 22-23KB
Sounds reasonnable

You can track the 'heap' value reported in the STATE message to see if you have a memory leak. The value should more or less remains constant in normal operations

**Reply by s-hadinger** [reply]
25kb is perfectly normal. It tends to be unstable below 12kb.

**Reply by ManfredBartz** [reply]
Ok thanks.
I didn't know the total is that low and what is normal for the platform.

**Reply by sfromis** [reply]
I consider 20KB as "getting low", a bit "uncomfortable", but not critical.
